### DISCUSSION TITLE
Why would a query not generate query parameters?

### DISCUSSION THREAD

**Discussion by katangafor** [trpc/trpc#4210] [Q&A]
I'm setting tRPC up in an existing react/express app, using [this](https://github.com/trpc/trpc/discussions/1860#discussioncomment-4406931) funky method, and so far typescript seems to be satisfied. I can define a procedure on the server, and (once the types are copied to the client app) I can call that procedure, with typescript enforcing the correct inputs.

But when I actually _make_ the call, the get request generated by tRPC doesn't have any query parameters, which of course immediately causes zod to throw an validation error in the server. My client is set up pretty much verbatim according to [this guide](https://trpc.io/docs/reactjs/setup) for setting tRPC up in react. 

I guess I'm just wondering if this sounds familiar to anybody. I feel like I'm missing some piece of configuration, but I can't figure out what it is.

**Comment by Nick-Lucas** [comment]
Are you able to share your code? Definitely sounds like a setup issue but hard to say

**Reply by katangafor** [reply]
On closer inspection, I that my regular ol' tanstack react-query provider is intercepting the queries and sending the problematic http requests.
I didn't notice this earlier, but there are two versions of the query that are getting sent: one that has the query parameters, and one that doesn't. I'm just guessing that my react-query provider is the culprit, but I'm not sure how to fix it.

My provider setup looks like this:

```typescript
import { useTrpcClient, trpc } from "trpc";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

const App = () => {
  const queryClient = new QueryClient();
  const trpcClient = useTrpcClient();

  return (
    <>
      <trpc.Provider client={trpcClient} queryClient={queryClient}>
        <QueryClientProvider client={queryClient}>
          <App />
        </QueryClientProvider>
      </trpc.Provider>
    </>
  );
};
``` 

`useTrpcClient` is just a hook that returns a client with a bearer token as the auth header:
```typescript
export const trpc = createTRPCReact<RouterTypes>();

export const useTrpcClient = () => {
  const token = useAppSelector((state) => state.user.token);

  return trpc.createClient({
    links: [
      httpLink({
        url: "http://localhost:8080/express/trpc",
        async headers() {
          return {
            Authorization: token ? `Bearer ${token}` : "",
          };
        },
      }),
    ],
  });
};
``` 

I noticed in the docs it says "If you already use React Query in your application, you should re-use the QueryClient and QueryClientProvider you already have." But I can't tell if the example that's directly below that is an example of re-using the QueryClient or not. Maybe this is my issue? Multiple QueryClients or something?

**Reply by Nick-Lucas** [reply]
```ts
const queryClient = new QueryClient();
```

Probably not related but you will need to memoise this, it contains your cache so you don't want to lose the cache on every render (though App might only render once right now)

You also definitely only need one QueryClient and Provider for react-query, if you have two you're may have problems.

This part of the setup looks otherwise fine though, react-query is just a caching and promise handling tool, tRPC is what creates the http requests.

**Reply by katangafor** [reply]
> Probably not related but you will need to memoise this, it contains your cache so you don't want to lose the cache on every render

Wow well ya learn something new every day!! I can't believe I never knew to memoize this. Thank you sir.
